tests/integration/golang/mlflow/run/create_test.go

```diff
@@ -128,27 +236,62 @@ func (s *CreateRunTestSuite) Test_Error() {
 			),
 		},
 		{
-			name: "CreateRunWithNotExistingExperiment",
+			name:      "CreateRunWithNotExistingNamespaceAndExistingExperimentID",
+			namespace: "not_existing_namespace",
 			request: request.CreateRunRequest{
-				ExperimentID: "1",
+				ExperimentID: fmt.Sprintf("%d", *experiment.ID),
 			},
 			error: api.NewResourceDoesNotExistError(
-				`unable to find experiment with id '1': error getting experiment by id: 1: record not found`,
+				`unable to find namespace with code: not_existing_namespace`,
+			),
+		},
+		{
+			name:      "CreateRunWithNotExistingExperiment",
+			namespace: "default",
+			request: request.CreateRunRequest{
+				ExperimentID: fmt.Sprintf("%d", nonExistingExperimentID),
+			},
+			error: api.NewResourceDoesNotExistError(
+				fmt.Sprintf(
+					`unable to find experiment with id '%d': error getting experiment by id: %d: record not found`,
+					nonExistingExperimentID,
+					nonExistingExperimentID,
+				),
+			),
+		},
+		{
+			name:      "CreateRunWithExistingNamespaceAndNotExistingExperiment",
+			namespace: "default",
```

Comment:
you can probably avoid `namespace: default`?

Reply:
i was running into some issues, possibly related to caching, where omitting the namespace did not select the default but rather the previously used namespace.

Reply:
okay "caching" because we are reusing the client across tests, it saved the namespace field. Changed this to always set namespace to test case field value, which clears it when omitted.